cEP-0015.md

```diff
@@ -0,0 +1,164 @@
+
+Errbot provides a pytest fixture `testbot` for testing purposes and to interact
+with the message queue. Extending the testbot and adding `inject_mock` method
+so that mocked objects can be injected into the plugin by passing a dictionary
+containing the `PluginName`, `field_name` and `Mocked Object`. The
+`inject_mock` method will inject a property on the plugin object and discard
+the assignments from the plugin itself.
+
+```
+def test_command(testbot):
+    mocks = {'PluginName': {'field_name1': fakeObj1,
+                                  'field_name2': fakeObj2}}
+              }
+    testbot.inject_mock(mocks)
+    assert 'something' in testbot.exec_command('!my_command')
+```
+
+## Configurability
+
+corobo has a potential to be used by other organizations for similar tasks like
+onboarding and automation. Currently, it is not configurable and many plugins
+are still very coala specific. Making it more configurable will allow other
+organizations to adapt corobo to cater their needs.
+
+Making existing coala specific plugins generic for other organizations and
+letting them configure the plugins as per their needs will ensure
+configurability.
+
+Errobot provides [plugin configuration](http://errbot.io/en/latest/user_guide/plugin_development/configuration.html)
+through the built-in `!config` command which can be used by other organizations
+to configure the plugins as per their needs.
+
+List of Plugins which are coala specific and can be generalized:
+
+1. `LabHub` plugin commands are meant to work for a coala specific team names
+   maintainers, developers, newcomers and coala repositories.
+
+   Issue Link: <https://github.com/coala/corobo/issues/382>
+
+2. `explain` currently uses hardcoded coala explainations. Configuring
+   sub-directories `explain/genric` and `explain/<org name>` so that
+   other organizations can include their culture specific explainations.
+
+3. `searchdocs` uses API_DOCS and USER_DOCS links which are constant for coala.
+    Other organizations will be able to initialize their own documentation
+    links as per their requirement.
+
+   Issue Link: <https://github.com/coala/corobo/issues/380>
+
+4. `community_stats` will be moved out from LabHub as a seperate plugin since
+   it can also be used by other organizations for analytics overview.
+
+   Issue Link: <https://github.com/coala/corobo/issues/361>
+
+### Configuration Sample
+
+```
+class LabHub(BotPlugin):
+    def get_configuration_template(self):
+        return {'GH_TEAM_NAMES': ['newcomers', 'developers', 'maintainers']}
+
+    @botcmd
+    def mycommand(self, mess, args):
+        # oh I need my TEAM !
+        gh_teams = self.config['GH_TEAM_NAMES']
+```
+
+```
+<BOT_PREFIX> plugin config LabHub {'GH_TEAM_NAMES': ['newbies', 'devs', 'admins']}
+```
+
+## New Features
+
+### Suggest newcomer issues
+
+This plugin will scrape issue links from GitHub issue tracker according to the
```

Comment:
Wouldnt it use the GitHub API using `IGitt`?

If it is necessary to use a scraper, you will need to explain why a proper API can not be used.